src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -167,18 +171,28 @@ public void Initialize()
                 }
 
                 _deadline = Clock.UtcNow.Add(timeout);
-                _cts = new CancellationTokenSource(timeout);
+
+                _timeoutCts = new CancellationTokenSource(timeout);
+                _timeoutCts.Token.Register(DeadlineExceeded, timeout);
+
+                _linkedCts = CancellationTokenSource.CreateLinkedTokenSource(_timeoutCts.Token, HttpContext.RequestAborted);
```

Comment:
Is it it worth having a cancellation token for the timeout? This could simplified as just a Timer that calls `DeadlineExceeded` which then aborts the HttpContext. `HttpContext.RequestAborted` could always be used.

@davidfowl

Reply:
You know how I feel about cancellation tokens and this isn't very pay for play. Does this always get allocated?

Reply:
It is pay for play. Cancellation tokens are only allocated if the client specifies a deadline.